src/plugins/routing.js

```diff
@@ -49,6 +55,10 @@ function switchLocalePathFactory (i18nPath) {
   const LOCALE_CODE_KEY = '<%= options.LOCALE_CODE_KEY %>'
 
   return function switchLocalePath (locale) {
+    if (STRATEGY === STRATEGIES.NO_PREFIX && locale !== defaultLocale) {
```

Comment:
Should be checking `locale !== this.$i18n.locale` instead. And missing `&& locale` condition.

Reply:
Also use `this[i18nPath].locale` instead.